src/encord_active/app/common/components/metric_summary.py

```diff
@@ -28,13 +30,63 @@
 )
 from encord_active.lib.metrics.utils import (
     MetricData,
+    MetricSchema,
     MetricScope,
     load_available_metrics,
 )
 
 _COLUMNS = MetricWithDistanceSchema
 
 
+def render_2d_metric_plots(metrics_data_summary: MetricsSeverity):
+    with st.expander("2D metrics view", True):
+        metric_selection_col, scatter_plot_col = st.columns([2, 5])
+        metric_names = [metric_name for metric_name in metrics_data_summary.metrics.keys()]
+        x_metric_name = metric_selection_col.selectbox("x axis", metric_names, index=0)
+        y_metric_name = metric_selection_col.selectbox("y axis", metric_names, index=1)
+        trend_selected = metric_selection_col.checkbox(
+            "Show trend",
+            value=True,
+            help="Draws a trend line to demonstrate the relationship between the two metrics.",
+        )
+
+        x_metric_df = metrics_data_summary.metrics[str(x_metric_name)].df[[MetricSchema.identifier, MetricSchema.score]]
+        x_metric_df.rename(columns={MetricSchema.score: f"{x_metric_name}"}, inplace=True)
+
+        y_metric_df = metrics_data_summary.metrics[str(y_metric_name)].df[[MetricSchema.identifier, MetricSchema.score]]
+        y_metric_df.rename(columns={MetricSchema.score: f"{y_metric_name}"}, inplace=True)
+
+        if len(x_metric_df.iloc[0][MetricSchema.identifier].split("_")) == len(
+            y_metric_df.iloc[0][MetricSchema.identifier].split("_")
+        ):
+            merged_metrics = pd.merge(x_metric_df, y_metric_df, how="inner", on=MetricSchema.identifier)
+        else:
+            x_changed, to_be_parsed_df = (
+                (True, x_metric_df.copy(deep=True))
+                if len(x_metric_df.iloc[0][MetricSchema.identifier].split("_")) == 4
+                else (False, y_metric_df)
+            )
+
+            to_be_parsed_df[[MetricSchema.identifier, "identifier_rest"]] = to_be_parsed_df[
+                MetricSchema.identifier
+            ].str.rsplit("_", n=1, expand=True)
```

Comment:
It seems like `to_be_parsed_df` can both be `x_metric_df` and `y_metric_df` but we only check the length of the key of `x_metric_df`. Could this potentially go wrong if `y_metric_df` is an image-level metric?

Reply:
In this else block, either `x_metric_df` or `y_metric_df` has length of four when split, so if it is not x, it should be y automatically.